src/store/modules/transactions/actions.js

```diff
@@ -25,8 +26,81 @@ export default {
   async get ({ state, commit, dispatch }, hash) {
     startLoading(dispatch, 'transactions/get')
 
-    commit('setTransactions', [])
+    const client = await ae
+    const transactions = await client.api.getTxs()
 
-    return endLoading(dispatch, 'transactions/get')
+    commit('setTransactions', transactions)
+
+    endLoading(dispatch, 'transactions/get')
+
+    return transactions
+  },
+
+  /**
+   * get transactions in the mempool
+   * @param {Object} state
+   * @param {Function} commit
+   * @param {Function} dispatch
+   * @param {String} hash
+   * @return {*}
+   */
+  async mempool ({ state, commit, dispatch }, hash) {
+    startLoading(dispatch, 'transactions/mempool')
+
+    const client = await ae
```

Comment:
Is it possible to put an initialized client into the state, to avoid `const client = await ae` duplication?

Reply:
@davidyuk okay

Reply:
In the Base app, I'm using [this plugin](https://github.com/aeternity/aepp-identity/blob/2028c64/src/store/plugins/initEpoch.js). In proposed implementation exception will happen if this action will be called before client initialization, but it won't if we will handle polling on the side of Vue store instead of polling function/mixin on the side of Vue components. As I see it, components should maintain a list of subscriptions in a global state (by "subscribe"/"unsubscribe" mutations) and display value of reactive variables from the store. Accordingly, the store should make polling according to a list of subscriptions, and update state displayed by components. Such architecture will make components simpler.